### ISSUE TITLE
is it possible to train yolo v8 on custom dataset of Pakistani currency to detect the value of currency  the link of dataset is   "https://www.kaggle.com/datasets/muhammadammarabid/pakistanicurrencydataset"    

### ISSUE THREAD

**Issue by niazikhan001** [ultralytics/ultralytics#9190]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

is it possible to train yolo v8 on custom dataset of Pakistani currency to detect the value of currency  the link of dataset is   please guide me https://www.kaggle.com/datasets/muhammadammarabid/pakistanicurrencydataset

**Comment by glenn-jocher** [comment]
@niazikhan001 absolutely, you can train YOLOv8 on a custom dataset of Pakistani currency to detect currency values! 🚀

First, ensure your dataset is in the correct format (YOLO format, COCO, etc.). If it's not, you might need to convert it. The dataset should include images of the currency and corresponding annotations (bounding boxes and labels).

Here's a simplified example of how you might start your training using YOLOv8:

1. Prepare your dataset and organize it into a structure that YOLOv8 expects. This typically includes splitting your data into train/val sets and creating a `.yaml` file describing your dataset paths and classes.

2. Create a `.yaml` file for your dataset. For example, `pak_currency.yaml`:

```yaml
train: /path/to/train/images
val: /path/to/val/images

nc: 5  # number of classes
names: ['10 Rupees', '20 Rupees', '50 Rupees', '100 Rupees', '500 Rupees']
```

3. Train YOLOv8 on your dataset:

```bash
yolo detect train data=pak_currency.yaml model=yolov8n.pt
```

Make sure to adjust paths, class names, and other configurations according to your dataset.

Happy training, and feel free to ask if you have more questions! 😊

**Comment by niazikhan001** [comment]
can you please guide me how i Organize your dataset into a structure that YOLOv8 expects and can it provide accurate predictions.  . Could you take a quick look at the dataset?

**Comment by niazikhan001** [comment]
@glenn-jocher  sir

**Comment by glenn-jocher** [comment]
@niazikhan001 certainly! To organize your dataset for YOLOv8, ensure your images and annotations are in separate folders, typically named `images` and `labels`. Inside these, you may have `train` and `val` subfolders to distinguish between your training and validation datasets. Annotations should be in YOLO format, where each `.txt` file corresponds to an image, detailing the bounding boxes and class IDs.

A quick look at the dataset shows it's well-suited for currency detection. Ensure each annotation file lists objects with: `<class_id> <x_center> <y_center> <width> <height>`, normalized to the image dimensions.

For accuracy, consider augmentations and tuning hyperparameters like learning rate. Remember, the more representative data you provide, the better YOLOv8 can learn! 🚀

```bash
yolo detect train data=path/to/your_dataset.yaml model=yolov8n.pt
```

This simple structure and command should kickstart your training. Let me know if you need further assistance!
